exporter/datadogexporter/translate_traces.go

```diff
@@ -550,16 +550,26 @@ func getSpanErrorAndSetTags(s pdata.Span, tags map[string]string) int32 {
 	if isError == errorCode {
 		extractErrorTagsFromEvents(s, tags)
 		// If we weren't able to pull an error type or message, go ahead and set
-		// these to the old defaults
+		// these to a reasonable default
 		if _, ok := tags[ext.ErrorType]; !ok {
-			tags[ext.ErrorType] = "ERR_CODE_" + strconv.FormatInt(int64(status.Code()), 10)
+			tags[ext.ErrorType] = "error"
 		}
 
 		if _, ok := tags[ext.ErrorMsg]; !ok {
 			if status.Message() != "" {
 				tags[ext.ErrorMsg] = status.Message()
 			} else {
-				tags[ext.ErrorMsg] = "ERR_CODE_" + strconv.FormatInt(int64(status.Code()), 10)
+				// look for useful http metadata if it exists and add that as a fallback for the error message
+				if tags[conventions.AttributeHTTPStatusCode] != "" {
+					if tags[conventions.AttributeHTTPStatusText] != "" {
+						tags[ext.ErrorMsg] = fmt.Sprintf("%s %s", tags[conventions.AttributeHTTPStatusCode], tags[conventions.AttributeHTTPStatusText])
+					} else {
+						tags[ext.ErrorMsg] = tags[conventions.AttributeHTTPStatusCode]
+					}
+				} else {
+					// as a fallback set the error to something straightforward
+					tags[ext.ErrorMsg] = "true"
```

Comment:
Do we have any equivalent on the Trace Agent to this fallback?

Reply:
So i looked into this a bit more and actually the Datadog Agent leaves these fields unset if they dont exist already, and adds a reasonable `error:true` in the UI in this case. So i have updated this PR to only add an `error.type` in the case that span.status.message or http status code information exists that allows us to generate a useful error message, otherwise we leave these fields unset (previously they'd default to err_code_2), and allow the Datadog UI to handle displaying a reasonable message (example of span with status of error but no type/msg info: https://a.cl.ly/Z4uKzKoL)